src/trials.jl

```diff
@@ -339,168 +355,194 @@ Base.show(io::IO, t::TrialRatio) = _show(io, t)
+    # artificial cases such as:  Trial(Parameters(), [3,4,5], [0,0,0], 0, 0)
+    fences = range(histmin, nextfloat(float(histmax)), length=histwidth)
+    bins = histogram_bindata(t.times, fences)
+    # Last bin is everything right of last fence, introduce a gap for printing:
+    _lastbin = pop!(bins)
+    push!(bins, 0, _lastbin)
+    if logbins
+        bins = log.(1 .+ bins)
     end
     hist = asciihist(bins, histheight)
-    hist[:,end-1] .= ' '
-    maxbin = maximum(bins)
-
-    delta1 = (histmax - histmin) / (histwidth - 1)
-    if delta1 > 0
-        medpos = 1 + round(Int, (histtimes[length(times) ÷ 2] - histmin) / delta1)
-        avgpos = 1 + round(Int, (mean(times) - histmin) / delta1)
-    else
-        medpos, avgpos = 1, 1
+    hist[:, end-1] .= ' '
+
+    avgpos = searchsortedlast(fences, avgtime)
+    medpos = searchsortedlast(fences, medtime)
+    q25pos = searchsortedlast(fences, quantile(t.times, 0.25)) # might be 0, that's OK
+    q75pos = searchsortedlast(fences, quantile(t.times, 0.75))
+
+    # Above the histogram bars, print markers for special ones:
+    printstyled(io, pad, "│", boxspace; color=boxcolor)
+    istop = maximum(filter(i -> i in axes(hist,2), [avgpos, medpos+1, q75pos]))
+    for i in axes(hist, 2)
+        i > istop && break
+        if i == avgpos
+            printstyled(io, "*", color=avgcolor, bold=true)
+        elseif i == medpos || 
+                (medpos==avgpos && i==medpos-1 && medtime<=avgtime) ||
+                (medpos==avgpos && i==medpos+1 && medtime>avgtime)
+            # Marker for "median" is moved one to the left if they collide exactly
+            # printstyled(io, "½", color=medcolor)
+            printstyled(io, "◑", color=medcolor)
+        elseif i == q25pos
+            # Quartile markers exist partly to explain the median marker, without needing a legend
+            # printstyled(io, "¼", color=:light_black)
+            printstyled(io, "◔", color=:light_black)
+        elseif i == q75pos
+            # printstyled(io, "¾", color=:light_black)
+            printstyled(io, "◕", color=:light_black)
+        else
+            print(io, " ")
+        end
     end
 
-    print(io, "\n")
     for r in axes(hist, 1)
-        print(io, "\n", pad, "  ")
+        println(io)
+        printstyled(io, pad, "│", boxspace; color=boxcolor)
+        istop = findlast(!=(' '), view(hist, r, :))
         for (i, bar) in enumerate(view(hist, r, :))
-            color = :default
-            if i == avgpos color = :green end
-            if i == medpos color = :blue end
-            printstyled(io, bar; color=color)
+            i > istop && break  # don't print trailing spaces, as they waste space when line-wrapped
+            if i == avgpos
+                printstyled(io, bar; color=avgcolor)
+                # If mean & median bars co-incide, colour the mean. Matches markers above.
+            elseif i == medpos
+                printstyled(io, bar; color=medcolor)
```

Comment:
When bars co-incide, this gives preference to the mean: green bar and no blue bar.

However, for the markers, at present the blue median marker will be offset slightly. Making it go missing seemed weird to me. But not sure this is the right decision.

The quartile markers will simply go missing if they coincide with any others.